### PR TITLE
Refactor corporation canpar

### PR DESCRIPTION
**Pull request by ventusignis**
Refactored base code so that `@game.corporation_parrable?(corporation, parrer)` is used instead of `corporation.can_par?(parrer)`
All of the game-specific oddities to corporation parring have been moved out of custom corporations into game code

Tagging @perwestling and @tysen because this affects 1824 and 1849 code

### CODE REVIEW

File: lib/engine/g_1856/corporation.rb [hunk 1]
```diff
@@ -49,13 +49,6 @@ def capitalization_type_desc
         CAPITALIZATION_STRS[@capitalization || _capitalization_type]
       end
 
-      def release_escrow!
```
Comment (ryandriskel) [inline lib/engine/g_1856/corporation.rb:52]: Intended change?

Comment (ventusignis) [reply lib/engine/g_1856/corporation.rb:52]: Not really, but I'm fine with making this change considering this method never gets called

Comment (ventusignis) [reply lib/engine/g_1856/corporation.rb]: I undid the change

Comment (ryandriskel) [reply lib/engine/g_1856/corporation.rb]: Sorry, I didn't mean for you to necessarily undo it, I just wanted to be sure you noticed it was included as it didn't seem related to the update. Sometimes I get distracted and change something and forget about it :D

File: lib/engine/game/base.rb [hunk 2]
```diff
@@ -1189,6 +1189,14 @@ def legal_tile_rotation?(_entity, _hex, _tile)
         true
       end
 
+      def corporation_parrable?(corporation, parrer)
```
Comment (tobymao) [inline lib/engine/game/base.rb:1192]: can we rename to can_par?(corporation, parrer)

File: lib/engine/corporation.rb [hunk 3]
```diff
@@ -25,7 +25,7 @@ class Corporation
     include Transfer
 
     attr_accessor :ipoed, :par_via_exchange, :max_ownership_percent, :float_percent, :capitalization, :max_share_price
-    attr_reader :companies, :min_price, :name, :full_name, :fraction_shares, :type
+    attr_reader :companies, :min_price, :name, :full_name, :fraction_shares, :type, :needs_token_to_par, :tokens
```
Comment (tobymao) [inline lib/engine/corporation.rb:28]: tokens is unnecessary here

File: lib/engine/g_1856/corporation.rb [hunk 4]
```diff
@@ -36,10 +36,6 @@ def can_buy?
         false
       end
 
-      def can_par?
-        self == @game.national ? false : super
-      end
-
       def par!
```
Comment (tobymao) [inline lib/engine/g_1856/corporation.rb:39]: don't need to do it for this PR, but this should be able to allow you to delete the game specific corp

File: lib/engine/game/g_1824.rb [hunk 5]
```diff
@@ -64,7 +64,7 @@ def init_corporations(stock_market)
         min_price = stock_market.par_prices.map(&:price).min
 
         self.class::CORPORATIONS.map do |corporation|
-          Engine::G1824::Corporation.new(
+          Engine::Corporation.new(
             min_price: min_price,
             capitalization: self.class::CAPITALIZATION,
```
Comment (tobymao) [inline lib/engine/game/g_1824.rb:69]: is this even necessary anymore

File: lib/engine/step/g_1856/buy_sell_par_shares.rb [hunk 6]
```diff
@@ -26,7 +26,7 @@ def process_par(action)
           share_price = action.share_price
           corporation = action.corporation
           entity = action.entity
-          raise GameError, "#{corporation} cannot be parred" unless corporation.can_par?(entity)
+          raise GameError, "#{corporation} cannot be parred" unless @game.corporation.parrable?(entity)
```
Comment (tobymao) [inline lib/engine/step/g_1856/buy_sell_par_shares.rb:29]: syntax error?

@game.can_par?(...)

File: lib/engine/game/base.rb (hunk 2 above)
Comment (ventusignis) [reply lib/engine/game/base.rb]: I've renamed it

File: lib/engine/corporation.rb (hunk 3 above)
Comment (ventusignis) [reply lib/engine/corporation.rb]: I've removed `tokens`, thanks!

**Review by tobymao — APPROVED** [review]

**Pull request merged**